src/Controller/PartyController.php

```diff
@@ -33,14 +33,24 @@ public function __construct(PartyHelper $partyHelper)
     /**
      * @Route("/add", name="party_add")
      */
-    public function addParty(CaseEntity $case, Request $request): Response
+    public function addParty(CaseEntity $case, PartyHelper $partyHelper, Request $request): Response
     {
         $this->denyAccessUnlessGranted('edit', $case);
 
         $party = new Party();
 
+        $preSelectType = $request->get('type');
+
+        // We use the assumptions that first mentioned party on list of parties is the 'main' party type e.g. lejer/udlejer.
+        $preSelectValue =
+            'counterparty' === $preSelectType
+                ? $partyHelper->getCounterpartyTypesByCase($case)[array_key_first($partyHelper->getCounterpartyTypesByCase($case))]
+                : null
+        ;
+
```

Comment:
This will not work if `$partyHelper->getCounterpartyTypesByCase` is empty. Use a variable for `$partyHelper->getCounterpartyTypesByCase($case)` and `reset` to get the first value.

Reply:
Updated.